src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
```

Comment:
```suggestion
  constructor(name) {
    super(name);
```
You need to take `name` in order to have access to it in `const checkName = animal => animal.name !== beast.name;`

Sorry I didn't see it in my first review.

Reply:
I didn't get this part actually. I mean, code works without name because we are extending the constructor to Animal. I didn't get what I am missing here. Can you explain?

Reply:
Actually, after reading some documentation if you do not take in new information or you don't need to modify any of the existing properties that were in parent class. You don't need a constructer at all. So you can just delete `constructor` and `super` completely.